### DISCUSSION TITLE
alsa - cannot execute alsamixer as user and no audio.

### DISCUSSION THREAD

**Discussion by NoSequel** [kisslinux/repo#370] [Q&A]
I just recently decided to install KISS Linux, and I'm running into an issue with ALSA. 

### Unable to execute `alsamixer` as my primary user.
Error: 
```ALSA lib confmisc.c:855:(parse_card) cannot find card '0'
ALSA lib conf.c:5111:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory
ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings
ALSA lib conf.c:5111:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name
ALSA lib conf.c:5111:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5599:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib control.c:1481:(snd_ctl_open_noupdate) Invalid CTL default
cannot open mixer: No such file or directory
```

However, whenever I try to execute `alsamixer` as root, it just executes perfectly fine.

### No audio at all
I'm not entirely sure if there's anything wrong with my kernel configuration, or if it's related to something else. I've tried many different solutions I was able to find online, but none of them worked so far.

`dmesg | grep snd` output: https://termbin.com/w3el
`lspci -k` output: https://termbin.com/p8xx

Any help would be appreciated!

**Comment by dylanaraps** [comment]
What are the permissions of the files in /dev/snd?

**Reply by NoSequel** [reply]
> What are the permissions of the files in /dev/snd?

https://termbin.com/beilv
Could this be wrong? I'm not entirely sure what it's supposed to be.

**Reply by NoSequel** [reply]
I changed the group of the files to `audio`, I'm now able to execute the `alsamixer` command, however still no audio. Video playback (in browser, at least) seems to be ridiculously fast as well (it's playing like 5 seconds a second).

**Reply by dylanaraps** [reply]
These files should be owned by the audio group. This is handled by the device manager at boot
time. If you have not enabled the device manager service this could be a potential cause for the
incorrect permissions.

See: https://kisslinux.org/wiki/service-management

**Reply by dylanaraps** [reply]
(The same goes for other things in /dev/)

**Reply by NoSequel** [reply]
Yes, I'm able to execute alsamixer already, but still no audio. `alsaconf` says "No supported PnP or PCI card found", so I'm not entirely sure whats wrong

**Reply by dylanaraps** [reply]
This sounds like it could be a kernel configuration issue.
Send your kernel .config, output from dmesg, hardware information
and anything else you may deem useful.

**Reply by NoSequel** [reply]
kernel .config (5.15.4) - https://termbin.com/yshx
`dmesg | grep snd` - https://termbin.com/hrfuh
`aplay -l` - https://termbin.com/xf2q

hardware:

- i7 10700k
- ASRock Z490 Pro 4 (info on ASRock website suggests it's an ALC 1200, however aplay -l mentions `ALCS1200A`)

**Reply by Xeno-Hypster** [reply]
Hi,

Just wanting to stick my head in to note that I just encountered this issue too, and not on a "new" install. I had to unpack a new chroot onto my existing system (unrelated) and after updating I am no longer able to access audio as a user. I've checked my permissions in /dev/snd and most files are owned by root, not audio (bar seq and timer). Root is able to get into alsamixer and running cmus under root gives me audio, but not under user.

I'm mentioning this mainly because I did not change kernel settings, nor updated or recompiled the kernel, legit just downloaded and unpacked the 2021.7-9 chroot and then did a kiss update.

-------------
Edit: Just wish to edit my response as I solved this by enabling mdev, I believe this was my bad and I'm unsure if this is related to NoSequel's issue.

**Reply by NoSequel** [reply]
I've been able to fix not being able to execute the `alsamixer`, but like I previously mentioned `alsaconf` still says "No supported PnP or PCI card found".
